napari/_qt/qt_main_window.py

```diff
@@ -586,6 +586,48 @@ def remove_dock_widget(self, widget):
         else:
             self._qt_window.removeDockWidget(widget)
 
+    def add_magic_function(self, function, magic=None, dock=None):
+        """Turn a function into a dock widget via magic gui.
+
+        Parameters
+        ----------
+        function : callable
+            Function that you want to add.
+        magic : dict, optional
+            Keyword arguments to the magicgui function that
+            can be used to specify widget.
+        dock : dict, optional
+            Keyword arguments to add_dock_widget function
+            that specify how the widget gets added.
+
+        Returns
+        -------
+        dock_widget : QtViewerDockWidget
+            `dock_widget` that can pass viewer events.
+        """
+        from magicgui import magicgui
+
+        if dock is None:
+            dock = {}
+
+        if 'name' not in dock:
+            function_name = function.__name__
+            function_name.replace('_', ' ')
+            dock['name'] = function_name
```

Comment:
```suggestion
        dock.setdefault('name', function.__name__.replace("_", " "))
```

(note also that line 615 here wouldn't actually replace the `function_name` variable...)

Reply:
not using dict anymore, so changed